namer/namer.cc

```diff
@@ -1300,7 +1404,11 @@ class SymbolDefiner {
         definedMethods.reserve(foundDefs.methods().size());
 
         for (auto ref : foundDefs.nonDeletableDefinitions()) {
-            defineNonDeletableSingle(ctx, ref);
+            // We say that the "owner" of fields declared in methods is the method itself.
+            // This is consistent with how we define the owner of contexts during treewalks.
+            if (ref.kind() != core::FoundDefinitionRef::Kind::Field) {
+                defineNonDeletableSingle(ctx, ref);
+            }
```

Comment:
ditto this, I might have misunderstood but I thought we were not going to need to do this anymore.